near-token-contract/src/lib.rs

```diff
@@ -93,10 +93,16 @@ impl Contract {
 
         // Set up access control.
         let super_admin = super_admin.unwrap_or_else(env::predecessor_account_id);
+
+        // Grant super admin role to the provided account.
         require!(
             contract.acl_init_super_admin(super_admin),
             "Failed to add factory as initial acl super-admin",
         );
+
+        // Grant MetadataUpdater role to the factory. This enables a trustless setup to set
+        // the metadata by any user. Super admin is responsible for adding a new user account
+        // with this role to provide icon and extra metadata.
```

Comment:
```suggestion
        // Grant MetadataUpdater role to the factory. This enables a trustless setup as
        // described in [`Self::update_metadata`].
```

I think the description of `update_metadata` is a bit easier to follow than the comment here.

Anyway, maybe we want to describe the idea only in one place? Then there's no risk of two descriptions getting out of sync.